src/query/storages/fuse/src/table_functions/clustering_information/clustering_information.rs

```diff
@@ -288,6 +340,142 @@ impl<'a> ClusteringInformation<'a> {
     }
 }
 
+fn get_min_max_stats(
```

Comment:
Seems that `get_min_max_stats` follows the way of range pruning, which is safe for range pruning , but **may** not be correct for calculate clustering information?



e.g.

~~~
create or replace table t (c string);

insert into t values ('abc'), ('abd');

insert into t values ('xyy'), ('xyz');

mysql> select * from clustering_information('default','t','(c)');
+-------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
| cluster_key | timestamp                  | total_block_count | constant_block_count | average_overlaps | average_depth | block_depth_histogram |
+-------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
| (c)         | 2024-06-13 02:45:11.740110 |                 2 |                    0 |              0.0 |           1.0 | {"00001":2}           |
+-------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
1 row in set (0.06 sec)
Read 1 rows, 448.00 B in 0.042 sec., 23.83 rows/sec., 10.43 KiB/sec.

mysql> select * from clustering_information('default','t','(substr(c,1))');
+-----------------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
| cluster_key           | timestamp                  | total_block_count | constant_block_count | average_overlaps | average_depth | block_depth_histogram |
+-----------------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
| (SUBSTRING(c FROM 1)) | 2024-06-13 02:45:11.740110 |                 2 |                    0 |              1.0 |           2.0 | {"00002":2}           |
+-----------------------+----------------------------+-------------------+----------------------+------------------+---------------+-----------------------+
1 row in set (0.06 sec)
Read 1 rows, 448.00 B in 0.030 sec., 33.45 rows/sec., 14.63 KiB/sec.
~~~

----------

Scanning all the relevant data from the table might be too intensive. Would sampling help?

Reply:
> Seems that `get_min_max_stats` follows the way of range pruning, which is safe for range pruning , but **may** not be correct for calculate clustering information?

> Scanning all the relevant data from the table might be too intensive. Would sampling help?

relate to the function domain
https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/scalars/string.rs#L666-L674

Reply:
Can I understand the trade offs this way?:

In general, functions used in ad-hoc cluster key expression may introduce some inaccuracies in the clustering information, as they are derived from the range index,  but as a trade-off, we don't need to scan a large amount of data, overall, it is cost-effective.

For some functions, relying on their properties (such as monotonicity), can help maintain more reasonable clustering information, though.